### PR TITLE
Update CMakePresets.json - Add Win64 preset

### PR DESCRIPTION
**Pull request by JeodC**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

This pull modifies the `win` preset to `win32` and adds the `win64` preset.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### Additional Comments

Important note: 32bit savegames are incompatible with 64bit builds for the moment.

### CODE REVIEW

**Comment by JeodC** [comment]
Will need to fix workflow as well.

File: CMakePresets.json [hunk 1]
```diff
@@ -18,6 +18,18 @@
...
+            "inherits": "defaults",
+            "generator": "Visual Studio 17 2022",
+            "condition": {
+                "type": "equals",
+                "lhs": "${hostSystemName}",
+                "rhs": "Windows"
+            },
+            "architecture": "x64",
+            "toolchainFile": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+        },
```
[the remark this answers is not in the document]
Comment (JeodC) [reply CMakePresets.json]: Fixed?

[the remark this answers is not in the document]
Comment (winterheart) [reply CMakePresets.json]: yes, now ok

**Comment by Lgt2x** [comment]
I would have preferred to keep the `win` preset as win32 as the default one for now in order too keep people's builds intact and smoothly transition to w64 later

**Pull request merged**